### ISSUE TITLE
Build error while fetching Python repository

### ISSUE THREAD

**Issue by ragequitting101** [google/mediapipe#2803]
**System information**
- OS Platform and Distribution: WSL1 on Windows 10 Home
- Installed using: WSL1
- MediaPipe version: Unsure, but downloaded on Nov 20, 2021
- Bazel version: 6.0.0-pre.20211110.1
- Android SDK and NDK versions (if android): SDK 32.0.0rc-1, NDK 21.4.7075529
- OpenCV version (if running on desktop): 4.2.0+dfsg-5

**Describe the problem**:
Hello!
I'm trying to install MediaPipe on Windows through WSL, but I faced some errors while following the steps on the installation page.
Since I'm using Windows to develop an Android application later, I followed the instructions here:
[https://google.github.io/mediapipe/getting_started/install.html#installing-on-windows-subsystem-for-linux-wsl](url)
I was able to follow along until step 7 (except for the installation of Windows ADB in step 2 where I was unable to start it, though I am not sure if I will use this feature).
In the final step to run the Hello World! example, I received the errors as shown in the code snippet below:

```
ragequitting101@LAPTOP-GTAQSH4K:/mnt/c/WINDOWS/system32$ cd mediapipe
ragequitting101@LAPTOP-GTAQSH4K:/mnt/c/WINDOWS/system32/mediapipe$ export GLOG_logtostderr=1
ragequitting101@LAPTOP-GTAQSH4K:/mnt/c/WINDOWS/system32/mediapipe$ bazel.exe run --define MEDIAPIPE_DISABLE_GPU=1 \
>     mediapipe/examples/desktop/hello_world:hello_world
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'bazel_skylib' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'pybind11' because it already exists.
INFO: Repository local_execution_config_python instantiated at:
  C:/windows/system32/mediapipe/WORKSPACE:409:14: in <toplevel>
  C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/tensorflow/workspace2.bzl:1085:19: in workspace
  C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/tensorflow/workspace2.bzl:85:27: in _tf_toolchains
  C:/users/david/_bazel_david/65xhtriv/external/tf_toolchains/toolchains/remote_config/configs.bzl:6:28: in initialize_rbe_configs
  C:/users/david/_bazel_david/65xhtriv/external/tf_toolchains/toolchains/remote_config/rbe_config.bzl:158:27: in _tensorflow_local_config
Repository rule local_python_configure defined at:
  C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl:275:41: in <toplevel>ERROR: An error occurred during the fetch of repository 'local_execution_config_python':
   Traceback (most recent call last):
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 209, column 22, in _create_local_python_repository
                _check_python_bin(repository_ctx, python_bin)
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 145, column 25, in _check_python_bin
                auto_config_fail("--define %s='%s' is not executable. Is it the python binary?" % (
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/remote_config/common.bzl", line 12, column 9, in auto_config_fail
                fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Configuration Error: --define PYTHON_BIN_PATH='C:\Users\david\AppData\Local\Programs\Python\Python310\python.exe' is not executable. Is it the python binary?
ERROR: C:/windows/system32/mediapipe/WORKSPACE:409:14: fetching local_python_configure rule //external:local_execution_config_python: Traceback (most recent call last):
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 209, column 22, in _create_local_python_repository
                _check_python_bin(repository_ctx, python_bin)
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 145, column 25, in _check_python_bin
                auto_config_fail("--define %s='%s' is not executable. Is it the python binary?" % (
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/remote_config/common.bzl", line 12, column 9, in auto_config_fail
                fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Configuration Error: --define PYTHON_BIN_PATH='C:\Users\david\AppData\Local\Programs\Python\Python310\python.exe' is not executable. Is it the python binary?
ERROR: Analysis of target '//mediapipe/examples/desktop/hello_world:hello_world' failed; build aborted: Configuration Error: --define PYTHON_BIN_PATH='C:\Users\david\AppData\Local\Programs\Python\Python310\python.exe' is not executable. Is
it the python binary?
INFO: Elapsed time: 0.830s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (1 packages loaded, 0 targets configured)
FAILED: Build did NOT complete successfully (1 packages loaded, 0 targets configured)
```

I have also tried adding the recommended --action flag from these pages and checking for the PYTHON_BIN_PATH variable, but still received the same error:
[https://google.github.io/mediapipe/getting_started/troubleshooting.html#missing-python-binary-path](url)
[https://github.com/google/mediapipe/issues/759](url)

![image](https://user-images.githubusercontent.com/80981496/142784695-bca77657-11a1-469a-958e-a8635d10e179.png)
```ragequitting101@LAPTOP-GTAQSH4K:/mnt/c/WINDOWS/system32$ cd mediapipe
ragequitting101@LAPTOP-GTAQSH4K:/mnt/c/WINDOWS/system32/mediapipe$ export GLOG_logtostderr=1
ragequitting101@LAPTOP-GTAQSH4K:/mnt/c/WINDOWS/system32/mediapipe$ bazel.exe build -c opt \
>   --define MEDIAPIPE_DISABLE_GPU=1 \
>   --action_env PYTHON_BIN_PATH="C:\Users\david\AppData\Local\Programs\Python\Python310\python.exe" \
>   mediapipe/examples/desktop/hello_world
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'bazel_skylib' because it already exists.
DEBUG: C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/repo.bzl:109:14:
Warning: skipping import of repository 'pybind11' because it already exists.
INFO: Build options --action_env and --compilation_mode have changed, discarding analysis cache.
INFO: Repository local_execution_config_python instantiated at:
  C:/windows/system32/mediapipe/WORKSPACE:409:14: in <toplevel>
  C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/tensorflow/workspace2.bzl:1085:19: in workspace
  C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/tensorflow/workspace2.bzl:85:27: in _tf_toolchains
  C:/users/david/_bazel_david/65xhtriv/external/tf_toolchains/toolchains/remote_config/configs.bzl:6:28: in initialize_r
be_configs
  C:/users/david/_bazel_david/65xhtriv/external/tf_toolchains/toolchains/remote_config/rbe_config.bzl:158:27: in _tensor
flow_local_config
Repository rule local_python_configure defined at:
  C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl:275:41: in <toplevel>
ERROR: An error occurred during the fetch of repository 'local_execution_config_python':
   Traceback (most recent call last):
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 20
9, column 22, in _create_local_python_repository
                _check_python_bin(repository_ctx, python_bin)
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 14
5, column 25, in _check_python_bin
                auto_config_fail("--define %s='%s' is not executable. Is it the python binary?" % (
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/remote_config/common.bzl", line 1
2, column 9, in auto_config_fail
                fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Configuration Error: --define PYTHON_BIN_PATH='C:\Users\david\AppData\Local\Programs\Python\Python310\pyt
hon.exe' is not executable. Is it the python binary?
ERROR: C:/windows/system32/mediapipe/WORKSPACE:409:14: fetching local_python_configure rule //external:local_execution_c
onfig_python: Traceback (most recent call last):
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 20
9, column 22, in _create_local_python_repository
                _check_python_bin(repository_ctx, python_bin)
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/py/python_configure.bzl", line 14
5, column 25, in _check_python_bin
                auto_config_fail("--define %s='%s' is not executable. Is it the python binary?" % (
        File "C:/users/david/_bazel_david/65xhtriv/external/org_tensorflow/third_party/remote_config/common.bzl", line 1
2, column 9, in auto_config_fail
                fail("%sConfiguration Error:%s %s\n" % (red, no_color, msg))
Error in fail: Configuration Error: --define PYTHON_BIN_PATH='C:\Users\david\AppData\Local\Programs\Python\Python310\pyt
hon.exe' is not executable. Is it the python binary?
ERROR: Analysis of target '//mediapipe/examples/desktop/hello_world:hello_world' failed; build aborted: Configuration Er
ror: --define PYTHON_BIN_PATH='C:\Users\david\AppData\Local\Programs\Python\Python310\python.exe' is not executable. Is
it the python binary?
INFO: Elapsed time: 1.961s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (37 packages loaded, 11 targets configured)
```

I even tried the steps from this page ([https://github.com/google/mediapipe/issues/724](url)), but nothing changed.

Please let me know what I can do to resolve the error, or if there was a step I may have overlooked.

Thank you!

**Comment by sgowroji** [comment]
Hi @ragequitting101, Can  you try downgrading the bazel to 3.7.2  and clean (bazel clean --expunge) before executing the build commands.

**Comment by keishatsai** [comment]
Hi for those whom use Windows 10 and search for the answer to this problem.

Adding **PYTHON_BIN_PATH** indeed solve the problem, but not adding it into environment variable (I had tried few times and it only cares whether we call "action_env" this argument or not).  Instead, we have to call "**--action_env**" when calling `bazel build`.... and **remember that you are you using Windows which means that you have to make a proper Windows path by using double slash to escape, "\\\\".**

To sum up:
`bazel build -c opt --define MEDIAPIPE_DISABLE_GPU=1 --action_env PYTHON_BIN_PATH="C:\\Users\\username\\AppData\\Local\\Programs\\Python\\Python37\python.exe" mediapipe/examples/desktop/hello_world`

Hope this help anyone who suffers from this issue.
